ghost/core/core/server/services/milestone-emails/service.js

```diff
@@ -26,14 +26,41 @@ module.exports = {
             } = require('@tryghost/milestone-emails');
             const config = require('../../../shared/config');
             const milestonesConfig = config.get('milestones');
-            const {GhostMailer} = require('../mail');
+            const hostSettings = config.get('hostSettings');
+            const events = require('../../lib/common/events');
 
-            const mailer = new GhostMailer();
             const repository = new InMemoryMilestoneRepository();
             const queries = new MilestoneQueries({db});
 
+            const slackEnabled = hostSettings?.milestones?.enabled && hostSettings?.milestones?.url;
+
+            if (slackEnabled) {
+                const {SlackNotifications} = require('@tryghost/slack-notifications');
+                const slackNotifications = new SlackNotifications({
+                    events,
+                    slackOptions: {unfurl: false},
+                    url: hostSettings.milestones.url,
+                    slackData: {
+                        username: 'Milestone Service',
+                        text: 'Milestone achieved!',
+                        attachments: [{
+                            fallback: 'Sorry, content cannot be shown.',
+                            title: '<%title%>',
+                            color: 'good',
+                            fields: [{
+                                title: '<%title%>',
+                                value: '<%value%>'
+                            }]
+                        }]
+                    }
+                });
+
+                slackNotifications.addEventTypes(['milestone.arr.achieved', 'milestone.members.achieved']);
```

Comment:
If we follow the same pattern as the `staff-service` we would not add the events externally, but would be handled inside of the `slack-notifications` service - I think this is preferable because you have a single place to see which events trigger notifications